desktop/mac/permission-requirements.md

```diff
@@ -7,15 +7,85 @@ redirect_from:
+```console
+$ docker run -p 127.0.0.1:80:80 nginx
+
+docker: Error response from daemon: Ports are not available: exposing port
+TCP 127.0.0.1:80 -> 0.0.0.0:0: failed to connect to /var/run/com.docker.vmnetd.sock:
+is vmnetd running?: dial unix /var/run/com.docker.vmnetd.sock: connect: connection
+refused.
+ERRO[0003] error waiting for container: context canceled
+```
+
+> **Note**
+>
+> The command may fail with the same error if user takes too long to authorize the prompt to start the helper process as it may timeout.
+
+The removal of the privileged helper process can be done in the typical way of removing launchd processes.
+
+```console
+$ ps aux | grep vmnetd                                                
+root             28739   0.0  0.0 34859128    228   ??  Ss    6:03PM   0:00.06 /Library/PrivilegedHelperTools/com.docker.vmnetd
+user             32222   0.0  0.0 34122828    808 s000  R+   12:55PM   0:00.00 grep vmnetd
+$ sudo launchctl unload -w /Library/LaunchDaemons/com.docker.vmnetd.plist
+Password:
+$ ps aux | grep vmnetd                                                   
+user             32242   0.0  0.0 34122828    716 s000  R+   12:55PM   0:00.00 grep vmnetd
+$ rm /Library/LaunchDaemons/com.docker.vmnetd.plist 
+$ rm /Library/PrivilegedHelperTools/com.docker.vmnetd 
+```
+
+### Ensuring `localhost` and `kubernetes.docker.internal` are defined
+
+On first run, Docker Desktop checks if `localhost` is resolved to `127.0.0.1`. In case the resolution fails, it prompts the user to allow adding the mapping to `/etc/hosts`. Similarly, when the Kubernetes cluster is installed, it checks that `kubernetes.docker.internal` is resolved to `127.0.0.1` and prompts the user to do so. 
+
+## Privileged Helper
+
+The privileged helper is started by `launchd` and runs in the background unless it is disabled at runtime as previously described. The Docker Desktop backend communicates with it over the UNIX domain socket `/var/run/com.docker.vmnetd.sock`. The functionalities it performs are: 
+- Installing and uninstalling symlinks in `/usr/local/bin`. 
+- Binding privileged ports that are less than 1024.
+- Ensuring `localhost` and `kubernetes.docker.internal` are defined in `/etc/hosts`. 
+- Securely caching the Registry Access Management policy which is read-only for the developer.
+- Performing some diagnostic actions, in particular gathering a performance trace of Docker itself.
+- Uninstalling the privileged helper.
 
 In version 4.11 and later of Docker Desktop for Mac you can avoid running the privileged helper service in the background by using the `--user` flag on the [install command](../install/mac-install.md#install-from-the-command-line). This results in `com.docker.vmnetd` being used for set up during installation and then disabled at runtime. In this case, the user is  not prompted to grant root privileges on the first run of Docker Desktop. Specifically, the `--user` flag:
```

Comment:
This is not accurate anymore, as we are no longer running the privileged helper service in the background. Using the --user flag performs the privileged configurations at install time, so that the user does not get the prompt on startup. 
So this paragraph does not belong to the "Privileged helper" section, but perhaps in its own section, something like "Performing privileged configurations at installation time".